File: Js/token_generator.js


function generateToken(len,lower,upper,num)

{
    
    
    let params = [lower,upper,num];
    let newParams = [];
    let token = [];
    

    const chars = [
        
        "abcdefghijklmnopqrstuvwxyz",
         "ABCDEFGHIJKLMNOPQRSTUVWXYZ",
         "0123456789"
    ];
    
    
    for (let i = 0; i < params.length; i++)
{
    
    switch (params[i])
    
{
        
        case true:
            newParams.push(chars[i]);
            break;
        
        default:
            // no default action actually
            

 }
    
    

}

    let newStr = newParams.join("");    
        
    for (let j = 0; j < len; j++)
    
{
    let rand = Math.floor(Math.random() * newStr.length);
    token[j] = newStr[rand];
}



switch(arguments.length){
    case 0: return generateToken(8,true,true,true);
    break;
    case 1: return generateToken(len,true,true,true);
    break;
    default: return token.join("");
}
}

console.log(generateToken())
console.log(generateToken(30))
console.log(generateToken(15,false,false,true))

File: Java/TokenGenerator.java
import java.util.ArrayList;


public class TokenGenerator {
	public static void main(String[] args) {
		System.out.println(generateToken(8,true,true,true));
	}
	
	public static String generateToken(int length, boolean uppercase, boolean lowercase, boolean number){

Boolean[] params = {uppercase,lowercase,number};

ArrayList<String> newParams = new ArrayList<>();

ArrayList<String> token = new ArrayList<>();

String[] chars = {
         "ABCDEFGHIJKLMNOPQRSTUVWXYZ",
        "abcdefghijklmnopqrstuvwxyz",
         "0123456789"
	    };
	    for(int i = 0; i < params.length; i++){
	        
	        if(params[i] == true){
				newParams.add(chars[i]);
	        }
	        
		}
		
		String newStr = String.join("",newParams);
		char[] newStr2 = newStr.toCharArray();

		for (int j = 0; j < length; j++){

			int rand = (int)  (Math.random() * newStr2.length);

			token.add(j, String.valueOf(newStr2[rand]));

		}
	    
	    
	    
	    return  String.join("",token);
	}
}